File: LatihanInheritance/Bus.cpp
#include "Kendaraan.hpp"

using namespace std;

class Bus : public Kendaraan {
private:
    string transportBrand;
    int releaseYear;
    int capacity;

public:
    Bus() : Kendaraan() {
        this->transportBrand = "XXX";
        this->releaseYear = 0;
        this->capacity = 0;
    }

    Bus(int transportID, string transportBrand, int releaseYear, int capacity) : Kendaraan(transportID) {
        this->transportBrand = transportBrand;
        this->releaseYear = releaseYear;
        this->capacity = capacity;
    }

    Bus(const Bus &bus) : Kendaraan(bus) {
        this->transportBrand = bus.transportBrand;
        this->releaseYear = bus.releaseYear;
        this->capacity = bus.capacity;
    }  

    Bus& operator= (const Bus &bus) {
        Kendaraan::operator=(bus);
        this->transportBrand = bus.transportBrand;
        this->releaseYear = bus.releaseYear;
        this->capacity = bus.capacity;
        return *this;
    }

    ~Bus() {}

    void printInfo() {
        cout << "Berikut informasi tentang bus ini: " << endl;
        cout << "ID Kendaraan: " << this->transportID << endl;
        cout << "Merk: " << this->transportBrand << endl;
        cout << "Tahun Keluaran: " << this->releaseYear << endl;
        cout << "Kapasitas: " << this->capacity << " orang" << endl;
    }

    int biayaSewa(int lamaSewa) {
        return 1000000 * lamaSewa;
    }
};

File: LatihanInheritance-Java/Bus.java
// Aria Judhistira / 13523112 / K02

public class Bus extends Kendaraan {
    private String merk;
    private int tahunKeluaran;
    private int kapasitas;

    public Bus() {
        super();
        this.merk = "XXX";
        this.tahunKeluaran = 0;
        this.kapasitas = 0;
    }
    public Bus(int kendaraanID, String merk, int tahun, int kapasitas) {
        super(kendaraanID);
        this.merk = merk;
        this.tahunKeluaran = tahun;
        this.kapasitas = kapasitas;
    }
    public Bus(Bus bus) {
        super(bus);
        this.merk = bus.merk;
        this.tahunKeluaran = bus.tahunKeluaran;
        this.kapasitas = bus.kapasitas;
    }

    @Override
    public void printInfo() {
        System.out.println("Berikut informasi bus ini: ");
        System.out.println("ID Kendaraan: " + kendaraanID);
        System.out.println("Merk: " + merk);
        System.out.println("Tahun Keluaran: " + tahunKeluaran);
        System.out.println("Kapasitas: " + kapasitas);
    }
    
    @Override
    public int biayaSewa(int lamaSewa) {
        return 1000000 * lamaSewa;
    }
}
